x-pack/plugins/search_inference_endpoints/public/components/all_inference_endpoints/tabular_page.tsx

```diff
@@ -5,28 +5,80 @@
  * 2.0.
  */
 
-import React, { useCallback } from 'react';
+import React, { useCallback, useEffect } from 'react';
 
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
 import { InferenceAPIConfigResponse } from '@kbn/ml-trained-models-utils';
 
 import { useTableData } from '../../hooks/use_table_data';
+import { FilterOptions } from './types';
+
+import { DeploymentStatusEnum } from './types';
 
 import { useAllInferenceEndpointsState } from '../../hooks/use_all_inference_endpoints_state';
 import { EndpointsTable } from './endpoints_table';
-import { TABLE_COLUMNS } from './table_columns';
+import { ServiceProviderFilter } from './filter/service_provider_filter';
+import { TaskTypeFilter } from './filter/task_type_filter';
+import { TableSearch } from './search/table_search';
+import { useTableColumns } from './render_table_columns/table_columns';
+import { useKibana } from '../../hooks/use_kibana';
 
 interface TabularPageProps {
   inferenceEndpoints: InferenceAPIConfigResponse[];
 }
 
 export const TabularPage: React.FC<TabularPageProps> = ({ inferenceEndpoints }) => {
-  const { queryParams, setQueryParams } = useAllInferenceEndpointsState();
+  const [searchKey, setSearchKey] = React.useState('');
+  const [deploymentStatus, setDeploymentStatus] = React.useState<
+    Record<string, DeploymentStatusEnum>
+  >({});
+  const { queryParams, setQueryParams, filterOptions, setFilterOptions } =
+    useAllInferenceEndpointsState();
+
+  const {
+    services: { ml },
+  } = useKibana();
+
+  const onFilterChangedCallback = useCallback(
+    (newFilterOptions: Partial<FilterOptions>) => {
+      setFilterOptions(newFilterOptions);
+    },
+    [setFilterOptions]
+  );
+
+  useEffect(() => {
+    const fetchDeploymentStatus = async () => {
+      const trainedModelStats = await ml?.mlApi?.trainedModels.getTrainedModelStats();
+      if (trainedModelStats) {
+        const newDeploymentStatus = trainedModelStats?.trained_model_stats.reduce(
+          (acc, modelStat) => {
+            if (modelStat.model_id) {
+              acc[modelStat.model_id] =
+                modelStat?.deployment_stats?.state === 'started'
+                  ? DeploymentStatusEnum.deployed
+                  : DeploymentStatusEnum.notDeployed;
+            }
+            return acc;
+          },
+          {} as Record<string, DeploymentStatusEnum>
+        );
+        setDeploymentStatus(newDeploymentStatus);
+      }
+    };
+
+    fetchDeploymentStatus();
```

Comment:
:x: Shouldn't we be handling errors from this? we would need to add a `.catch()` to do this. If the `await ml?.mlApi?.trainedModels.getTrainedModelStats()` we'll have an unhandled promise rejection in the browser which would be bad.

Reply:
Updated.